.github/workflows/ci.yml

```diff
@@ -0,0 +1,87 @@
+name: CI
+
+on:
+  push:
+    branches: [master, staging]
+  pull_request:
+    branches: [master, staging]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CHROME_BIN: /usr/bin/google-chrome
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  test:
+    name: Checks & Build Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 12
+      - name: Install Dependenices
+        run: npm install
+      - name: Run Test with CodeCov
+        env:
+          CHROME_BIN: /usr/bin/google-chrome
+        run: npm run test:codecov
+      - name: Run Lint & Prettier Tests
+        run: |
+          npm run lint:type
+          npm run prettier:check
+      - name: Build staging
+        run: npm run build:stage
+      - name: build production
+        run: npm run build:prod
+      - name: Code coverage test
+        if: success()
```

Comment:
I think `success()` is effectively the default condition for a step in a job.

Any specific reason you are specifying it here? is there some nuance or other behavior you are trying to suppress?

Reply:
I added it to ensure all previous steps are successful to run this job. But it's been some time so I'll give it a go without it to see.

Reply:
jobs are supposed to fail fast by default, so if any step fails it will skip all subsequent steps unless they have `if: always()`

Reply:
for reference here is where we use the codecov action in the godtools android app:
https://github.com/CruGlobal/godtools-android/blob/develop/.github/workflows/build.yml#L80-L81

Reply:
Thanks for the reference.

Reply:
I'm unsure why I added the `success()` reading the docs; it's like you said, if a step fails, it skips all subsequent steps.

Reply:
maybe just a copy and paste because it was listed under `after_success` in travis 🤷